scala/private/rule_impls.bzl

```diff
@@ -559,10 +567,15 @@ def _lib(ctx, non_macro_lib):
     )
 
 def scala_library_impl(ctx):
-  return _lib(ctx, True)
+  scalac_provider = ctx.attr._scalac[_ScalacProvider]
+  return _lib(ctx, scalac_provider.default_compile_classpath,
+              scalac_provider.default_runtime_classpath, True)
 
 def scala_macro_library_impl(ctx):
-  return _lib(ctx, False)  # don't build the ijar for macros
+  scalac_provider = ctx.attr._scalac[_ScalacProvider]
+  return _lib(ctx, scalac_provider.default_macro_classpath,
+              scalac_provider.default_macro_classpath,
```

Comment:
do macros need reflect at runtime as well?

Reply:
Macros generally would only need reflect at runtime if they generate code that uses reflect. In that case, I think they should add reflect as an export since it unusual.